### PR TITLE
[TensorExpr] Fast sigmoid for LLVM

### PR DESCRIPTION
**Pull request by zheng-xq**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39717 [TensorExpr] Fast sigmoid for LLVM**

Differential Revision: [D21949849](https://our.internmc.facebook.com/intern/diff/D21949849)

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/llvm_codegen.cpp [hunk 1]
```diff
@@ -322,19 +322,24 @@ void LLVMCodeGenImpl::emitWrapper(const std::vector<llvm::Type*>& params) {
...
       }
+    } else if (v->op_type() == kSigmoid) {
+      ScalarType stype = v->dtype().scalar_type();
+      if (stype == ScalarType::Float) {
+        return fast_sigmoid(v->param(0));
+      }
     }
     // TODO: fast exp
     // TODO: fast erf
     // TODO: fast sigmoid
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/llvm_codegen.cpp:341]: Nit: remove the finished "TODO" please.

File: torch/csrc/jit/tensorexpr/codegen.h [hunk 2]
```diff
@@ -182,6 +186,11 @@ TORCH_API std::unique_ptr<CodeGen> CreateCodeGen(
     const std::vector<CodeGen::BufferArg>& params,
     at::Device device = at::kCPU);
 
+class TORCH_API GenericIntrinsicsExpander : public IRMutator {
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/codegen.h:189]: Nit: we're creating an entire class (+ derived from it) just to redefine a single function. We could've get away with just a single free-standing function here.

**Review by ZolotukhinM — CHANGES_REQUESTED** [review]
Overall looks good, but please add some tests for this change.

File: torch/csrc/jit/tensorexpr/codegen.h (hunk 2 above)
Comment (zheng-xq) [reply torch/csrc/jit/tensorexpr/codegen.h:189]: This class and function is invoked implicity through accept_mutator. Unless we come up with more magic wrappers here, there is no easy way to call into this in our current infrastructure with a free standing function.

File: torch/csrc/jit/tensorexpr/llvm_codegen.cpp (hunk 1 above)
Comment (zheng-xq) [reply torch/csrc/jit/tensorexpr/llvm_codegen.cpp]: Done

**Review by ZolotukhinM — APPROVED** [review]
Looks good if the tests are there!

**Pull request closed without merging**